hypothesis-python/src/hypothesis/internal/conjecture/shrinking/floats.py

```diff
@@ -65,18 +65,13 @@ def run_step(self):
         # change that would require shifting the exponent while not changing
         # the float value much.
 
-        for g in [math.floor(self.current), math.ceil(self.current)]:
-            self.consider(g)
-
         if self.consider(int(self.current)):
             self.debug("Just an integer now")
             self.delegate(Integer, convert_to=int, convert_from=float)
             return
 
-        m, n = self.current.as_integer_ratio()
-        i, r = divmod(m, n)
+        # Try dropping precision bits
 
-        # Now try to minimize the top part of the fraction as an integer. This
-        # basically splits the float as k + x with 0 <= x < 1 and minimizes
-        # k as an integer, but without the precision issues that would have.
-        self.call_shrinker(Integer, i, lambda k: self.consider((i * n + r) / n))
```

Comment:
This call accomplishes nothing? Probably, `i` should be replaced by `k` in the lambda; but I'm not convinced it's worthwhile even so.

Reply:
On balance, I'm happy to take it out: as you note it didn't work before, and all our tests keep passing without it.  In the very unlikely event that users report any degredation, we can consider restoring a repaired version.